src/components/common/ColonyHome/ColonyDomainSelector/ColonyDomainSelector.tsx

```diff
@@ -87,7 +87,7 @@ const ColonyDomainSelector = ({
   return (
     <Form<FormValues>
       defaultValues={{
-        filteredDomainId: String(filteredDomainId),
```

Comment:
This was a bug. 

We should be storing the domain id as a number, as the domain select components are using numbers and so a === comparison wasn't returning true when it should've.